programs/bpf_loader/src/lib.rs

```diff
@@ -180,9 +180,6 @@ pub fn load_program_from_account(
         return Err(InstructionError::IncorrectProgramId);
     }
 
-    let (programdata_offset, deployment_slot) =
```

Comment:
Is this code movement necessary?
Bit concerned about error priorities here.

Reply:
Yes, since we don't load the actual program account, the `get_programdata_offset_and_depoyment_offset` returns an error, since `state()` is not valid.

Reply:
Looking through the code, `get_programdata_offset_and_depoyment_offset()` returns `InvalidAccountData` in all error scenarios. So, moving the code after cache lookup should not change the error codes (as the cache lookup failure returns the same error code).